src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -208,5 +208,30 @@ public string Value
         ///   Returns <see cref="JsonValueKind.String"/>
         /// </summary>
         public override JsonValueKind ValueKind { get => JsonValueKind.String; }
+
+        /// <summary>
+        ///   Converts the text value of this instance, which should encode binary data as base-64 digits, to an equivalent 8-bit unsigned <see cref="byte"/> array.
+        ///   The return value indicates wether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="byte"/> array equivalent of the text contained in this instance,
+        ///   if the conversion succeeded.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if text was converted successfully; othwerwise returns <see langword="false"/>.
+        /// </returns>
+        internal bool TryGetBytesFromBase64(out byte[] value)
+        {
+            try
+            {
+                value = Convert.FromBase64String(_value);
```

Comment:
Use `Convert.TryFromBase64String` instead of `try-catch`.

Reply:
This was used in #41464 ([this method](https://github.com/dotnet/corefx/pull/41464/files#diff-503f2d7e6ad23069afcb9b5e735ce9fb)) but was scrapped in this implementation since that required me to declare `byte[] value`, which in itself would require extensive testing for the declaration. See [this method](https://github.com/dotnet/corefx/blob/dcc9f626eeb0751d4ba4499f623aac42d2e95c10/src/Common/src/CoreLib/System/Convert.cs#L2815-L2868) in Convert for an example. 

I could miss something that makes using `Convert.TryFromBase64String` an easy implementation, but as I see it I would need to declare `byte[] value` with the proper size if I should use this method. This might be what is required, but I thought it would not be since this is only an internal method.

Reply:
`Convert.TryFromBase64String` takes a `Span<byte>` as destination. So one could use a big enough destination, then slice it with `bytesWritten` and with `ToArray` create the result.

It's about safing the (unnecessary) exception-logic, which will result in better perf and avoids allocations (from the exceptional path).

So I would write this method as
```c#
internal bool TryGetBytesFromBase64(out byte[] value)
{
    // we decode string -> byte, so the resulting length will
    // be /4 * 3, but to be on the safe side, just use the string length
    int bufferSize = _value.Length;

    byte[] arrayToReturnToPool = null;
    try
    {
        Span<byte> buffer = bufferSize <= 256
            ? stackalloc byte[bufferSize]
            : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);

        if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
        {
            value = buffer.Slice(0, bytesWritten).ToArray();
            return true;
        }
        else
        {
            value = default;
            return false;
        }
    }
    finally
    {
        if (arrayToReturnToPool != null)
        {
            ArrayPool<byte>.Shared.Return(arrayToReturnToPool);
        }
    }
}
```

Reply:
Maybe the `arrayToReturnToPool` should be cleared before returning to the pool.
I don't remember what policy is followed here.

Reply:
@ahsonkhan you probably know whether `arrayToReturnToPool` should be cleared.

I pushed changes in line with your suggestion. However, I changed `Debug.Assert(_value.Length > 0);` to an `if-statement`, because I could not get the tests to run with the prior statement, even though it builds. This was only a problem when I added test cases where `_value.Length` was lower than 1.

The implementation is tested and passed all tests on my local implementation. Let me know if the `Debug.Assert` is necessary and I'll add it once again. If the tests still fail to run I'll let you know why.

Reply:
I added `Debug.Assert` in the assumption that for this internal method `_value` will have a length > 0, but I didn't flow through the code, so maybe my assumption was wrong.
With your change it is on the safe side anyway.

Reply:
The span-based Convert.TryFromBase64String isn't available on netfx/netstandard2.0 (only available on netcoreapp2.1 and higher and netstandard2.1).
https://apisof.net/catalog/System.Convert.TryFromBase64String(String,Span%3CByte%3E,Int32)

That's why the netfx CI leg is failing (https://dev.azure.com/dnceng/public/_build/results?buildId=376897):
> System\Text\Json\Node\JsonString.cs(243,29): error CS0117: 'Convert' does not contain a definition for 'TryFromBase64String'

If we want to use the span-based one, do it under an #ifdef on `BUILDING_INBOX_LIBRARY` and fall-back to the existing string-based Convert API for older platforms, similar to https://github.com/dotnet/corefx/blob/342faad8745aaef3f07fc4931435eabfdcc0f978/src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L652-L656:
https://apisof.net/catalog/System.Convert.FromBase64String(String)

> you probably know whether `arrayToReturnToPool` should be cleared.

In general, the recommendation is to clear arrays from the pool, particularly if they contain sensitive data that shouldn't be made visible to the caller or any other user of the pool that might rent the same array. Other, similar uses of the arraypool within `JsonDocument`, clear the slice of the array before returning. **So, for consistency and to be extra safe, let's clear it (I don't see perf being of utmost importance here either).** If performance ends up being important here, a case can be made for leaving it uncleared since the contents are being returned to the caller anyway and it's a single token. But I would start off with clearing it.

Reply:
I chose to keep the implementation as suggested by @gfoidl under `#if BUILDING_INBOX_LIBRARY `. I readded the old implementation for older platforms. However, this still suffers from the problems pointed out by @gfoidl. Should I reimplement this solution to not include `try-catch`? I have no solution on top of my head but I could try to find something if it is needed.